tests/Unit/Generator/EntityRequestSchemaBuilderTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\LoyaltyCorp\ApiDocumenter\Unit\Generator;
+
+use cebe\openapi\spec\Schema;
+use Doctrine\Common\Persistence\Mapping\ClassMetadataFactory;
+use Doctrine\Common\Persistence\ObjectManager;
+use LoyaltyCorp\ApiDocumenter\Generator\EntityRequestSchemaBuilder;
+use LoyaltyCorp\ApiDocumenter\Generator\Exceptions\UnsupportedClassException;
+use Tests\LoyaltyCorp\ApiDocumenter\Stubs\Externals\ORM\ManagerRegistryStub;
+use Tests\LoyaltyCorp\ApiDocumenter\TestCases\TestCase;
+use Tests\LoyaltyCorp\ApiDocumenter\Unit\Generator\Fixtures\PublicProperties;
+
+/**
+ * @covers \LoyaltyCorp\ApiDocumenter\Generator\EntityRequestSchemaBuilder
+ */
+final class EntityRequestSchemaBuilderTest extends TestCase
+{
+    /**
+     * Tests that supports listens to its internal $skipEntities array.
+     *
+     * @return void
+     *
+     * @throws \cebe\openapi\exceptions\TypeErrorException
+     */
+    public function testBuildUnsupported(): void
+    {
+        $managerRegistry = new ManagerRegistryStub();
+        $builder = new EntityRequestSchemaBuilder(
+            $managerRegistry,
+            null,
+            [PublicProperties::class]
+        );
+
+        $this->expectException(UnsupportedClassException::class);
+        $this->expectExceptionMessage('The class "Tests\LoyaltyCorp\ApiDocumenter\Unit\Generator\Fixtures\PublicProperties" is not supported by this schema builder.'); // phpcs:ignore
```

Comment:
For these kind of messages can you use sprintf with the ::class because if we refactor we need to change them all when the message is still actually correct.